engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/aggregate/TableAggregationFactory.scala

```diff
@@ -0,0 +1,126 @@
+import pl.touk.nussknacker.engine.flink.api.process.FlinkCustomStreamTransformation
+import pl.touk.nussknacker.engine.flink.table.aggregate.TableAggregationFactory._
+
+object TableAggregationFactory {
+
+  val groupByParamName: ParameterName            = ParameterName("groupBy")
+  val aggregateByParamName: ParameterName        = ParameterName("aggregateBy")
+  val aggregatorFunctionParamName: ParameterName = ParameterName("aggregator")
+  private val outputVarParamName: ParameterName  = ParameterName(OutputVar.CustomNodeFieldName)
+
+  private val groupByParam: ParameterExtractor[LazyParameter[AnyRef]] with ParameterCreatorWithNoDependency =
+    ParameterDeclaration.lazyMandatory[AnyRef](groupByParamName).withCreator()
+
+  private val aggregateByParam: ParameterExtractor[LazyParameter[AnyRef]] with ParameterCreatorWithNoDependency =
+    ParameterDeclaration.lazyMandatory[AnyRef](aggregateByParamName).withCreator()
+
+  private val aggregatorFunctionParam = {
+    val aggregators = Aggregator.allAggregators.map(a => FixedExpressionValue(s"'${a.name}'", a.name))
+    ParameterDeclaration
+      .mandatory[String](aggregatorFunctionParamName)
+      .withCreator(
+        modify = _.copy(editor = Some(FixedValuesParameterEditor(FixedExpressionValue.nullFixedValue +: aggregators)))
+      )
+  }
+
+}
+
+class TableAggregationFactory
+    extends CustomStreamTransformer
+    with SingleInputDynamicComponent[FlinkCustomStreamTransformation] {
+
+  override type State = Nothing
+
+  override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])(
+      implicit nodeId: NodeId
+  ): ContextTransformationDefinition = {
+    case TransformationStep(Nil, _) =>
+      NextParameters(
+        parameters = groupByParam
+          .createParameter() :: aggregateByParam.createParameter() :: aggregatorFunctionParam.createParameter() :: Nil,
+        errors = List.empty,
+        state = None
+      )
+    case TransformationStep(
+          (`groupByParamName`, groupByParam) ::
+          (`aggregateByParamName`, aggregateByParam) ::
+          (`aggregatorFunctionParamName`, DefinedEagerParameter(aggregatorName: String, _)) :: Nil,
+          _
+        ) =>
+      val outName = OutputVariableNameDependency.extract(dependencies)
+
+      val selectedAggregator = Aggregator.allAggregators
+        .find(_.name == aggregatorName)
+        .getOrElse(throw new IllegalStateException("Aggregator not found. Should be invalid at parameter level."))
+
+      val aggregatorOutputType = selectedAggregator.outputType(aggregateByParam.returnType)
+
+      val aggregateByTypeErrors = selectedAggregator.inputTypeConstraint match {
+        case Some(typeConstraint) =>
+          if (!aggregateByParam.returnType.canBeSubclassOf(typeConstraint)) {
+            List(
+              // TODO: this is a different message from other aggregators - choose one and make it consistent for all
```

Comment:
Not sure about this: Right now for other aggregate components its `Could not create aggregate-sliding: Invalid aggregate type: String(str), should be: Number`. I think this here message is better but it adds inconsistency. Should I change it in streaming aggregates to make it consistent?

Reply:
Unfortunately it requires quite a lot of changes and imo it would make this PR bloated. I propose treating it the same as making the labels consistent and get back to it when it's more of a priority

Reply:
Unfortunately it requires quite a lot of changes and imo it would make this PR bloated. I propose treating it the same as making the labels consistent and get back to it when it's more of a priority